crates/bevy_transform/src/components/transform.rs

```diff
@@ -373,100 +373,100 @@ impl Transform {
     ///
     /// More precisely, the [`Transform::rotation`] produced will be such that:
     /// * applying it to `main_axis` results in `main_direction`
     /// * applying it to `secondary_axis` produces a vector that lies in the half-plane generated by `main_direction` and
     /// `secondary_direction` (with positive contribution by `secondary_direction`)
     ///
-    /// [`Transform::look_to`] is recovered, for instance, when `main_axis` is `Vec3::NEG_Z` (the [`Transform::forward`]
-    /// direction in the default orientation) and `secondary_axis` is `Vec3::Y` (the [`Transform::up`] direction in the default
+    /// [`Transform::look_to`] is recovered, for instance, when `main_axis` is `Dir3::NEG_Z` (the [`Transform::forward`]
+    /// direction in the default orientation) and `secondary_axis` is `Dir3::Y` (the [`Transform::up`] direction in the default
     /// orientation). (Failure cases may differ somewhat.)
     ///
     /// In some cases a rotation cannot be constructed. Another axis will be picked in those cases:
-    /// * if `main_axis` or `main_direction` is zero, `Vec3::X` takes its place
-    /// * if `secondary_axis` or `secondary_direction` is zero, `Vec3::Y` takes its place
+    /// * if `main_axis` or `main_direction` fail converting to `Dir3` (e.g are zero), `Dir3::X` takes their place
+    /// * if `secondary_axis` or `secondary_direction` fail converting, `Dir3::Y` takes their place
     /// * if `main_axis` is parallel with `secondary_axis` or `main_direction` is parallel with `secondary_direction`,
     /// a rotation is constructed which takes `main_axis` to `main_direction` along a great circle, ignoring the secondary
     /// counterparts
     ///
     /// Example
     /// ```
-    /// # use bevy_math::{Vec3, Quat};
+    /// # use bevy_math::{Dir3, Vec3, Quat};
     /// # use bevy_transform::components::Transform;
     /// # let mut t1 = Transform::IDENTITY;
     /// # let mut t2 = Transform::IDENTITY;
-    /// t1.align(Vec3::X, Vec3::Y, Vec3::new(1., 1., 0.), Vec3::Z);
-    /// let main_axis_image = t1.rotation * Vec3::X;
+    /// t1.align(Dir3::X, Dir3::Y, Vec3::new(1., 1., 0.), Dir3::Z);
+    /// let main_axis_image = t1.rotation * Dir3::X;
     /// let secondary_axis_image = t1.rotation * Vec3::new(1., 1., 0.);
     /// assert!(main_axis_image.abs_diff_eq(Vec3::Y, 1e-5));
     /// assert!(secondary_axis_image.abs_diff_eq(Vec3::new(0., 1., 1.), 1e-5));
     ///
-    /// t1.align(Vec3::ZERO, Vec3::Z, Vec3::ZERO, Vec3::X);
-    /// t2.align(Vec3::X, Vec3::Z, Vec3::Y, Vec3::X);
+    /// t1.align(Vec3::ZERO, Dir3::Z, Vec3::ZERO, Dir3::X);
+    /// t2.align(Dir3::X, Dir3::Z, Dir3::Y, Dir3::X);
     /// assert_eq!(t1.rotation, t2.rotation);
     ///
-    /// t1.align(Vec3::X, Vec3::Z, Vec3::X, Vec3::Y);
+    /// t1.align(Dir3::X, Dir3::Z, Dir3::X, Dir3::Y);
     /// assert_eq!(t1.rotation, Quat::from_rotation_arc(Vec3::X, Vec3::Z));
     /// ```
```

Comment:
Would be nice if the other methods also gotten an example like this, showcasing how you can mix&match vectors and directions to improve discoverability of the fact that both Dir3 and Vec3 are valid arguments, for new users.

Not a necessary change though.